### PR TITLE
More Boilerplate + setting up dev env

### PR DESCRIPTION
**Pull request by kentkwu**
So I did a couple of things here:
1. Added ESLint (airbnb, mostly) and flow typechecking
2. Fleshed out Redux with Thunk Middleware
3. Implemented basic signup page

I should be able to move quite a bit quicker after this. Setting up the configs and stuff took a bit.

If you follow the instructions in the Readme, you should have live flow checking and linting in Nuclide (if you decide to use Nuclide)

Most of this is still just boilerplate. When we get back to school we can dive into the implementation details of the stuff here.

screenshot 2016-08-22 02 30 45

### CODE REVIEW

File: src/actions/SignupActions.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  description: 'Something went wrong',
+})
+
+export const signupUser = (
+  firstName: string,
+  lastName: string,
+  email: string,
+  password: string
+) => (dispatch: Function) => {
+  request.post('http://localhost:5000/signup')
```
Comment (dsiah) [inline src/actions/SignupActions.js:23]: Is this port your oberon server? It doesn't match the port that npm start runs this application

File: src/components/signup/Signup.react.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import Header from '../shared/Header.react.js'
+import SignupContainer from '../../containers/SignupContainer.react.js'
+
+const Signup = () => (
+  <div>
+    <Header />
+    <div className={'contentContainer'}>
```
Comment (dsiah) [inline src/components/signup/Signup.react.js]: nit: does this need to be a string evaluated? Typically I can write this without the curly braces

File: src/components/signup/SignupForm.react.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      firstName,
+      lastName,
+      password,
+      email,
+    },
+    signupSubmit,
+  }: {
+    fields: Object,
+    signupSubmit: Function,
+  } = props
```
Comment (dsiah) [inline src/components/signup/SignupForm.react.js:20]: clarify: what is going on with this syntax (specifically lines 8 - 20)

File: src/static/App.css [hunk 4]
```diff
@@ -1,24 +1,23 @@
...
 }
 
-.App-intro {
-  font-size: large;
-}
-
-@keyframes App-logo-spin {
-  from { transform: rotate(0deg); }
-  to { transform: rotate(360deg); }
+#signupForm {
```
Comment (dsiah) [inline src/static/App.css]: nit: typically its better to use a class selector over an id selector. There are a few reasons as far as how this affects cascading styles, but for the most part it's a preference thing that has been enforced especially due to CSS lint

https://css-tricks.com/a-line-in-the-sand/

**Comment by dsiah** [comment]
1. So most of the files are not 100% covered by flow, is that the plan going forward or will there be some other policy for flow coverage.
2. I think I understand what a thunk is, but why do we need this middleware?
3. Where do you write the validation rules for the form now that we are using redux-form does it go in the reducer?

File: src/actions/SignupActions.js (hunk 1 above)
Comment (kentkwu) [reply src/actions/SignupActions.js:23]: Yeah npm run start serves the client on 3000

File: src/components/signup/SignupForm.react.js (hunk 3 above)
Comment (kentkwu) [reply src/components/signup/SignupForm.react.js:20]: This is what the flow check looks like for an object. It ensures that fields is an Object and signupSubmit is a func

**Comment by kentkwu** [comment]
1. It won't be 100% because we are using libraries that aren't flowchecked. If you import something that isn't typechecked by flow, then flow can't make any guaranteed about typing where you use that import. That being said, we should have flow as much as we can in every file.
2. It makes it much easier to perform async action dispatches. You can also conditionally dispatch actions with thunk. In actions, if you see any action that returns a function, it is a thunk.
3. That's on my to do list. You don't put it in the reducer though you just pass the validation function into SignupForm.react.js at like 83

**Pull request merged**